src/go/k8s/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -666,6 +699,16 @@ type LoadBalancerStatus struct {
 	corev1.LoadBalancerStatus `json:""`
 }
 
+// NodePoolStatus describes the status of the NodePool.
+type NodePoolStatus struct {
+	CurrentReplicas int32 `json:"currentReplicas,omitempty"`
+	Replicas        int32 `json:"replicas,omitempty"`
+	ReadyReplicas   int32 `json:"readyReplicas,omitempty"`
+	// Indicates that a nodePool's pods are restarting.
+	// +optional
+	Restarting bool `json:"restarting"`
```

Comment:
Isn't this implied by `CurrentReplicas < Replicas`? If this field is added to appease TF, could you please note that?

Reply:
this is to comply with current operator v1, which has a restarting field top-level in status. 

top-level restarting is computed from all NP's restarting